raw_decoder/raw_decoder.go

```diff
@@ -0,0 +1,317 @@
+package raw_decoder
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"golbat/grpc"
+	"golbat/pogo"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type RawDecoder interface {
+	GetProtoDataFromHTTP(http.Header, []byte) (*ProtoData, error)
```

Comment:
Do alternative ways of receiving data make sense to put in the same interface?
Surely we just get the data into a standard for (ProtoData I guess) and then pass on to a common processor?

Reply:
Yeah, this doesn't feel right to me, either. Not sure what I was smoking that night. Starting to refresh myself with what I did here.. since this was 1.5mo ago. :) I've been running with this patch this whole time, though. I'll fix this.